docs/design/2019-11-05-index-advisor.md

```diff
@@ -0,0 +1,185 @@
+
+# Proposal: Index Advisor
+
+- Author(s): [Zou Huan](https://github.com/zouhuan1215) 
+- Last updated:  Nov. 5th, 2019
+- Discussion at: 
+
+## Abstract
+
+This proposal aims at designing a new DBA tool -- index advisor, which automatically recommends indexes for input workload. Having added these recommend indexes to tables, the overall execution time of input workload is supposed to decrease. Current proposal only considers single-column index recommendation.
+
+
+## Background
+
+Appropriate indexes can speed up query execution a lot. However, in real-world applications, determining proper indexes for a database according to a typical workload purely by hand, is an excessively challenging task. When deciding reasonable indexes for a large-scale database, this task becomes even more tough as one database may contain dozens of tables in which each table may have dozens of columns, whereas in the typical workload thousands of queries are involved in. Fortunately, we already have an excellent tool -- TiDB SQL optimizer that can help us get through this tough task. The main purpose of this proposal is to utilize existing TiDB SQL optimizer to automatically recommend suitable indexes for database given the workload.
+
+To illustrate how can SQL optimizer help us determine suitable indexes, here is an intuitive example: 
```

Comment:
`us` mean `index advisor` or users? Maybe you can make it more clear.

Reply:
*Fortunately, we already have an excellent tool -- TiDB SQL optimizer which can be utilized to get through this tough task*